scripts/test/Muon/plotting_canvas/plotting_canvas_presenter_test.py

```diff
@@ -185,6 +185,78 @@ def test_create_tiled_plot_correctly_handles_empty_key_list(self):
+        self.presenter._options_presenter.check_autoscale_state = mock.Mock()
+        self.presenter._options_presenter.check_autoscale_state.return_value = True
+        self.presenter._options_presenter.get_selection = mock.Mock()
+        self.presenter._options_presenter.get_selection.return_value = [1,2,3]
+        self.presenter._view.get_axis_limits = mock.Mock()
+        self.presenter._view.get_axis_limits.return_value = 0, 1, 0, 1
+        self.presenter._handle_autoscale_y_axes()
+        mock_autoscale_y_axes.assert_called_once_with()
+
+    @mock.patch(
+    'Muon.GUI.Common.plot_widget.plotting_canvas.plotting_canvas_presenter.PlottingCanvasPresenter.get_workspace_info')
+    @mock.patch(
+    'Muon.GUI.Common.plot_widget.plotting_canvas.plotting_canvas_presenter.PlottingCanvasPresenter.get_plot_axes')
+    def test_autoscale_y_axes_calculates_correct_yerr(self,mock_get_plot_axes,mock_get_workspace_info):
+        self.presenter._options_presenter.get_errors = mock.Mock()
+        self.presenter._options_presenter.get_errors.return_value = True
+        mock_get_workspace_info.return_value = [WorkspacePlotInformation(workspace_name="test",index = 0,
+                                                                        axis = 0, normalised = True,
+                                                                        errors = True, label = 'Test Workspace')]
+
+        mock_get_plot_axes.return_value =[MockAxis(0,10)]
+        x_data = range(0,51)
+        y_data = [x**2 for x in x_data]
+        e_data = range(0,51)
+        test = CreateWorkspace(x_data,y_data,e_data)
+        self.presenter.autoscale_y_axes()
+        self.presenter._view.autoscale_y_axes.assert_called_once_with(10.0)
+
+    @mock.patch(
+    'Muon.GUI.Common.plot_widget.plotting_canvas.plotting_canvas_presenter.PlottingCanvasPresenter.get_workspace_info')
+    @mock.patch(
+    'Muon.GUI.Common.plot_widget.plotting_canvas.plotting_canvas_presenter.PlottingCanvasPresenter.get_plot_axes')
+    def test_autoscale_selected_y_axis_calculates_correct_yerr(self,mock_get_plot_axes,mock_get_workspace_info):
+        self.presenter._options_presenter.get_errors = mock.Mock()
+        self.presenter._options_presenter.get_errors.return_value = True
+        mock_get_workspace_info.return_value = [WorkspacePlotInformation(workspace_name="test", index=0,
+                                                                         axis=0, normalised=True,
+                                                                         errors=True, label='Test Workspace')]
+
+        mock_get_plot_axes.return_value = [MockAxis(0, 10)]
+        x_data = range(0, 51)
+        y_data = [x ** 2 for x in x_data]
+        e_data = range(0, 51)
+        test = CreateWorkspace(x_data, y_data, e_data)
+        self.presenter.autoscale_selected_y_axis(0)
+        self.presenter._view.autoscale_selected_y_axis.assert_called_once_with(0,10.0)
```

Comment:
I dont see how this checks we have the correct y error. Could you please explain it to me?

Reply:
The MockAxis returns the x limits in which the data is to be used  which in this case is 0 to 10 and in the x range the maximum error is 10 which is passed to the view via the autoscale_selected_y_axis function along with the axis number 0 which was passed into the presenter autoscale_selected_y_axis.

Reply:
can we change it so that the max error does not match the x axis value?

Reply:
if not it will use the largest error in the workspace and not what is shown on the plot. the x limits returned ensures the search for max error is restricted to the data shown on graph.

Reply:
can you rewrite this test giving the numbers descriptive variables names to make it more clear what is going on?